### PR TITLE
Transforms setup

### PR DESCRIPTION
**Pull request by cgmillette**
- Removes redundant classes from cross_lane_ops (migrated into the backend files)
- Added static loop unrolling for dpp, swizzle, blend and permute
- Introduced some helper functions to transforms_impl.hpp :
-- extractEven = form a new vector from even elements of the given vector
-- extractOdd = form new vector from odd elements of the given vector
-- concat = form new vector from concatenation of two given vectors
-- unpackLo2/4/8 = form a new vector by alternating low elements from two given vectors
-- unpackHi2/4/8 = form a new vector by alternating high elements from two given vectors
- Optimized unpackLo/Hi by using packed b32 sized vectors
- Implemented unpackLoHi as optimized combinations of unpackLo in low regs and unpackHi in hi regs
- Implemented aos_soa_16xk_b32 as an example.

### CODE REVIEW

File: library/include/rocwmma/internal/blend_impl.hpp [hunk 1]
```diff
@@ -105,8 +107,12 @@ namespace rocwmma
                 template <typename DataT>
                 ROCWMMA_DEVICE static inline DataT exec(DataT src0, DataT src1)
                 {
+                    static_assert(sizeof(DataT) == sizeof(uint32_t), "Inputs must be 32 bit");
                     uint32_t const mask = MaskCtrl::maskCtrl();
-                    return (src1 & mask) | (src0 & ~mask);
+                    reinterpret_cast<uint32_t&>(src0)
```
Comment (dlangbe) [inline library/include/rocwmma/internal/blend_impl.hpp:112]: What is the need for these reinterpret_casts after the assertion of 32bit size?

File: library/include/rocwmma/internal/transforms_impl.hpp [hunk 2]
```diff
@@ -93,97 +96,394 @@ namespace rocwmma
...
 
+    ///
+    /// AOS -> SOA : Transform from inline VW to ortho VW
+    ///
+    template <typename DataT, uint32_t VecSize, uint32_t... Idx>
+    ROCWMMA_DEVICE constexpr static inline auto extractEven(VecT<DataT, VecSize> const& v,
+                                                            detail::SeqT<Idx...>)
+    {
+        static_assert(sizeof...(Idx) == VecSize / 2u, "Index count must be half the vector size");
+        return VecT<DataT, VecSize / 2u>{get<Idx * 2>(v)...};
```
Comment (dlangbe) [inline library/include/rocwmma/internal/transforms_impl.hpp:107]: Do we want to use bit shifting for multiplying and dividing VecSize by 2?

**Review by dlangbe — APPROVED** [review]

File: library/include/rocwmma/internal/blend_impl.hpp (hunk 1 above)
Comment (cgmillette) [reply library/include/rocwmma/internal/blend_impl.hpp:112]: Bitwise ops require integer types, this one we are expecting b32 size

File: library/include/rocwmma/internal/transforms_impl.hpp (hunk 2 above)
Comment (cgmillette) [reply library/include/rocwmma/internal/transforms_impl.hpp:107]: Logically, makes sense but for readability's I didn't manually do this. Will consider if it's affecting compile time significantly

**Pull request merged**